torch/onnx/symbolic_opset11.py

```diff
@@ -80,6 +80,13 @@ def index_put(g, self, indices_list_value, values, accumulate=False):
         ]
         index = g.op("Concat", *indices_list, axis_i=-1)
     else:
+        bool_inp = list(index.node().inputs())[0]
+        if bool_inp.type().scalarType() == 'Bool':
+            if values.node().kind() == 'onnx::Constant':
```

Comment:
why the requirement on constant?

From below looks like trying to get rank info. You can achieve that through `values.type().dim()`, remember to check `values.type()` is not None.